tools/kicad/kicad_functions/src/vdiv.rs

```diff
@@ -18,35 +21,57 @@ fn parse_series(str: &str) -> EvalexprResult<&'static RSeries> {
     }
 }
 
-// evalexpr is not smart enough to distinguish identifiers on its own
-fn unique_identifiers(e: &Node) -> usize {
+fn resistor_identifiers(e: &Node) -> usize {
+    let re = Regex::new(r"^R[1-9][0-9]*$").unwrap();
     let mut set = HashSet::new();
-    e.iter_variable_identifiers().for_each(|i| {
-        set.insert(i);
-    });
+    e.iter_variable_identifiers()
+        .filter(|i| re.is_match(i)) // Match only R? identifiers
+        .for_each(|i| {
+            set.insert(i);
+        });
     set.len()
```

Comment:
Why is the `HashSet` used here? Couldn't this be just the following?
```rust
fn resistor_identifiers(e: &Node) -> usize {
    let re = Regex::new(r"^R[1-9][0-9]*$").unwrap();
    e.iter_variable_identifiers()
        .filter(|i| re.is_match(i)) // Match only R? identifiers
        .count()
}
```

Reply:
Say you have parsed an expression of the form `R2 = R1 * R1`. What `iter_variable_identifiers()` is going to output is simply all variable identifiers of the parsed node tree in order, i.e. the list `R2, R1, R1`. The `HashSet` is used to keep track of already encountered identifiers for a relatively simple way to avoid counting a given identifier more than once.